airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -180,19 +182,21 @@ public static ServerRunnable getServer(ServerFactory apiFactory) throws Exceptio
             .getAndInitialize();
     final JobPersistence jobPersistence = new DefaultJobPersistence(jobDatabase);
 
-    createDeploymentIfNoneExists(jobPersistence);
+    UUID deploymentId = createDeploymentIfNoneExists(jobPersistence);
 
-    // must happen after deployment id is set
+    // if no workspace exists, we create one so the user starts out with a place to add configuration.
+    Optional<StandardWorkspace> newWorkspace = createNewWorkspaceIfNoneExists(configRepository);
+
+    // must happen after the workspace is created, because the tracking identity depends on it.
     TrackingClientSingleton.initialize(
         configs.getTrackingStrategy(),
-        new Deployment(DeploymentMode.OSS, jobPersistence.getDeployment().orElseThrow(), configs.getWorkerEnvironment()),
+        new Deployment(DeploymentMode.OSS, deploymentId, configs.getWorkerEnvironment()),
```

Comment:
Pass in the deployment ID explicitly so there is no implicit dependency.

I want to do this for the workspace as well when initializing the tracking client. Unfortunately, I cannot do that cleanly.